middle_end/flambda/simplify/simplify_expr.rec.ml

```diff
@@ -70,4 +70,22 @@ let simplify_expr dacc expr ~down_to_up =
                 Name_occurrences.print free_names
                 Expr.print expr
             end;
+
+            let code_size_uacc = UA.size uacc in
+            let denv = UA.creation_dacc uacc |> DA.denv in
+            let code_size =
+              Code_size.expr_size expr ~find_code:(DE.find_code denv)
+            in
+
+            if not (Code_size.equal code_size_uacc code_size)
```

Comment:
Actually this will only be on when the "FREE_NAMES" environment variable exists. I don't think it's a good idea to put it in the invariants as calling [expr_size] on every expr is extremely slow. This check is here mainly to test the PR by making sure the new computation (incrementally during simplify) is yielding the same size as the full computation and should be removed later on.

Reply:
Ah, I didn't see the call to `getenv`.